src/audio/volume/volume.c

```diff
@@ -157,14 +158,31 @@ static struct comp_dev *volume_new(struct sof_ipc_comp *comp)
 	schedule_task_init(&cd->volwork, SOF_SCHEDULE_LL, SOF_TASK_PRI_MED,
 			   vol_work, dev, 0, 0);
 
-	/* set the default volumes */
+	/* Set the default volumes. If IPC sets min_value or max_value to
+	 * not-zero, use it. Otherwise set to internal limits and notify
+	 * ramp step calculation about assumed range with the ange set to
+	 * zero.
+	 */
+	if (vol->min_value || vol->max_value) {
+		cd->vol_min = MAX(vol->min_value, VOL_MIN);
+		cd->vol_max = MIN(vol->max_value, VOL_MAX);
+		cd->vol_ramp_range = vol->max_value - vol->min_value;
```

Comment:
Why do we assume that full range goes to ramp_range? Wouldn't it be better to have it 0 by default in all cases and set explicitly if you want something else?

Reply:
Ah, maybe the variable name is not that good. The code assumes that the topology defined ramp length happens for largest possible transition. The largest transition is grabbed to volume code usage usage here. Was this your concern? If yes, maybe name volume_max_delta or something like that is better perhaps?

Reply:
My concern is why result will be different for default case and case when I explicitly pass default values. Le'ts assume that VOL_MIN is 0 and VOL_MAX is 1. When I don't set them, fw sets them for me to 0:1 and ramp is 0, but when I set them explicitly to 0:1 I get ramp 1. It may be confusing.

Reply:
I use the value 0 to indicate old kernel driver mode for volume_set_chan() where the default max transition is assumed to be 0.0 to 1.0. If the range is set to nonzero the step size is scaled with the actual known range.

Reply:
So, would a comment add here be sufficient?

Reply:
@singalsu Yes, thanks.

Reply:
OK!